controllers/common/common_status_update.go

```diff
@@ -0,0 +1,394 @@
+// Copyright (c) 2023 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package common
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	clusterv1beta1 "open-cluster-management.io/api/cluster/v1beta1"
+	appsv1 "open-cluster-management.io/multicloud-operators-subscription/pkg/apis/apps/placementrule/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	policiesv1 "open-cluster-management.io/governance-policy-propagator/api/v1"
+	policiesv1beta1 "open-cluster-management.io/governance-policy-propagator/api/v1beta1"
+)
+
+// RootStatusUpdat update Root policy status with bound decisions and placements
+func RootStatusUpdate(c client.Client, rootPolicy *policiesv1.Policy) error {
+	placements, decisions, err := GetClusterDecisions(c, rootPolicy)
+	if err != nil {
+		log.Info("Failed to get any placement decisions. Giving up on the request.")
+
+		return errors.New("could not get the placement decisions")
+	}
+
+	cpcs, cpcsErr := CalculatePerClusterStatus(c, rootPolicy, decisions)
+	if cpcsErr != nil {
+		// If there is a new replicated policy, then its lookup is expected to fail - it hasn't been created yet.
+		log.Error(cpcsErr, "Failed to get at least one replicated policy, but that may be expected. Ignoring.")
+	}
+
+	err = c.Get(context.TODO(),
+		types.NamespacedName{
+			Namespace: rootPolicy.Namespace,
+			Name:      rootPolicy.Name,
+		}, rootPolicy)
+	if err != nil {
+		log.Error(err, "Failed to refresh the cached policy. Will use existing policy.")
+	}
+
+	// make a copy of the original status
+	originalCPCS := make([]*policiesv1.CompliancePerClusterStatus, len(rootPolicy.Status.Status))
+	copy(originalCPCS, rootPolicy.Status.Status)
```

Comment:
Where is the copy used?

Reply:
I copy and pasted, it is used at rootpolicy_controller

Reply:
I had a question too lol why copy there?

Reply:
Yeah, it'd be nice to reuse this function in the root policy controller and return the copy for it to call `cleanUpOrphanedRplPolicies` with. Then we wouldn't have code duplication.